includes/Classifai/Providers/Provider.php

```diff
@@ -195,6 +195,60 @@ public function render_select( $args ) {
 			echo '<br /><span class="description">' . wp_kses_post( $args['description'] ) . '</span>';
 		}
 	}
+
+	/**
+	 * Renders the checkbox group for 'Generate alt text' setting.
+	 *
+	 * @param array $args The args passed to add_settings_field.
+	 */
+	public function render_auto_caption_fields( $args ) {
+		$setting_index = $this->get_settings();
+
+		$default_value = '';
+
+		if ( isset( $setting_index['enable_image_captions'] ) ) {
+			if ( ! is_array( $setting_index['enable_image_captions'] ) && '1' === $setting_index['enable_image_captions'] ) {
+				$default_value = 'alt';
+			} elseif ( ! is_array( $setting_index['enable_image_captions'] ) && 'no' === $setting_index['enable_image_captions'] ) {
+				$default_value = '';
+			}
+		}
+
+		$checkbox_options = array(
+			'alt'         => esc_html__( 'Alt text', 'classifai' ),
+			'caption'     => esc_html__( 'Image caption', 'classifai' ),
+			'description' => esc_html__( 'Image description', 'classifai' ),
+		);
+
+		foreach ( $checkbox_options as $option_value => $option_label ) {
+			if ( ! is_array( $setting_index['enable_image_captions'] ) ) {
+				$default_value = '1' === $setting_index['enable_image_captions'] ? 'alt' : '';
```

Comment:
These two lines throw a notice if you visit the page before the options are saved. To reproduce:

* Delete the `classifai_computer_vision` option from the database
* Visit the options page
* A notice will be thrown

Saving the option then caused different errors, continue from above:
* save the options without making any changes
* notices are thrown on line 233 for each item in the loop:
   * Undefined index: alt
   * Undefined index: caption
   * Undefined index: description